lib/coverband/adapters/s3_report_writer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Coverband
+  module Adapters
+    ###
+    # TODO: this is currently a html report writer
+    # this should support writing coverage the current method should be fine
+    # to write every report to S3 and sum them later or use the 2 pass
+    # method we do for redis if in a background thread
+    ###
+    class S3ReportWriter
```

Comment:
Is this class an adaptor like RedisStore?

Reply:
yeah I feel like I might move the S3 html reporter to utils and make a pure S3 adapter

Reply:
Wouldn't this need a save_coverage method to be an adaptor?

Reply:
yeah let me move this out of adapters until I actually implement those pieces and instead have what this does live in Utils.

Reply:
moved